flang/include/flang/Parser/parse-tree.h

```diff
@@ -3872,6 +3872,19 @@ struct OpenMPDeclareTargetConstruct {
   std::tuple<Verbatim, OmpDeclareTargetSpecifier> t;
 };
 
+struct OmpDeclareMapperSpecifier {
+  TUPLE_CLASS_BOILERPLATE(OmpDeclareMapperSpecifier);
+  std::tuple<std::optional<Name>, TypeSpec, Name> t;
```

Comment:
Parsing the type as `TypeSpec` would make it easier to print meaningful errors.  Otherwise, if the user provides an intrinsic type, the parser would just flag a syntax error, which may be confusing.

Reply:
That makes sense, thanks :)